script/update_containers.sh

```diff
@@ -76,26 +76,58 @@ else
   HOSTS="$*"
 fi
 
-if [[ $DEPLOY_ENV == beta ]]
-then
-  pause=90
-else
-  pause=30
-fi
-
-last_host=${HOSTS/* }
-
 for host in $HOSTS
 do
+  for service in $SERVICES
+  do
+    container="$service-$DEPLOY_ENV"
+    echo "$host: Putting $container into maintenance mode..."
+
+    ssh "$host" \
+      sudo -H -S -- \
+        /root/docker-server-configs/scripts/set_service_maintenance.sh \
+          enable \
+          "$container" \
+          5000
+  done
+
+  # Production value of DETERMINE_MAX_REQUEST_TIME
+  sleep 60
+
   echo "$host: Updating containers..."
   ssh "$host" \
     sudo -H -S -- \
       /root/docker-server-configs/scripts/update_services.sh \
         $DEPLOY_ENV $SERVICES
-  if [[ $host != "$last_host" ]]
-  then
-    sleep $pause
-  fi
+
+  for service in $SERVICES
+  do
+    container="$service-$DEPLOY_ENV"
+    http_status=0
+    while [[ $http_status -ne 200 ]]
+    do
+      http_status=$(
+        ssh "$host" \
+          docker exec "$container" \
+            curl -I -s -o /dev/null -w "%{http_code}" 'http://localhost:5000'
+      )
+      if [[ $http_status -eq 200 ]]
+      then
+        break
+      else
+        echo "$host: Waiting for $container to come back up..."
+        sleep 1
+      fi
+    done
```

Comment:
It should be noted that this script can fail indefinitely for various reasons:
* `$host` is currently not reachable or not correctly configured in ssh,
* user isn’t granted permissions to run `docker` or `docker` service is down,
* the MusicBrainz Server in the remote container never comes back up for some reason.

This is not necessarily a problem since this script is meant to be run manually thus monitored.

Reply:
I didn't test the suggested change but it appears logical.